webpack.config.js

```diff
@@ -186,6 +186,55 @@ function lorisModule(mname, entries) {
   };
 }
 
+const plugins = [
+  new CopyPlugin({
+    patterns: [
+      {
+        from: path.resolve(__dirname, 'node_modules/react/umd'),
+        to: path.resolve(__dirname, 'htdocs/vendor/js/react'),
+        force: true,
+        globOptions: {
+          ignore: ['react.profiling.min.js'],
+        },
+        filter: async (path) => {
+          const file = path.split('\\').pop().split('/').pop();
+          const keep = [
+            'react.development.js',
+            'react.production.min.js',
+          ];
+          return keep.includes(file);
+        },
+      },
+      {
+        from: path.resolve(__dirname, 'node_modules/react-dom/umd'),
+        to: path.resolve(__dirname, 'htdocs/vendor/js/react'),
+        force: true,
+        filter: async (path) => {
+          const file = path.split('\\').pop().split('/').pop();
+          const keep = [
+            'react-dom.development.js',
+            'react-dom.production.min.js',
+          ];
+          return keep.includes(file);
+        },
+      },
+    ],
+  }),
+];
+
+mode == 'development' && plugins.push(new ESLintPlugin({
```

Comment:
I don't think this is the right way of determining this. Your config.xml shouldn't affect your build. Whether you're running `make` or `make dev` should determine the build environment.

Reply:
@laemtl @driusan one method would be modify the Makefile to contain:
````
all: VERSION js
	composer install --no-dev

# If anything changes, re-generate the VERSION file
VERSION: .
	tools/gen-version.sh

phpdev:
	composer install

js:
	npm ci
	mode=production npm run compile

jsdev:
	npm ci
	mode=development npm run compile

jslatest: clean
	rm -rf package-lock.json
	rm -rf modules/electrophysiology_browser/jsx/react-series-data-viewer/package-lock.json
	npm install
	npm run compile

dev: VERSION phpdev jsdev
...
````

then inside webpack.config.js use `const mode = process.env.mode;` to obtain whether "development" or "production" and problem solved.

Reply:
@maltheism @driusan Let me know if this works for both of you.

Reply:
@driusan Those line set the mode to production/dev, which is implicitly done by passing   node-env to webpack (package.json):
 ```
"compile": "webpack --node-env=development",
 "build": "webpack --node-env=production",
```